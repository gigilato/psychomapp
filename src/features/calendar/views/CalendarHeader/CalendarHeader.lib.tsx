import { eachDayOfInterval, endOfWeek, startOfWeek } from '$libs/dates'
import { colors, heights } from '$theme'

export const date = new Date()
const from = startOfWeek(date)
const to = endOfWeek(date)

export const interval = eachDayOfInterval({ start: from, end: to })
export const textColor = 'text-white-classic'

export const height = heights.calendar
export const gradient = [colors.secondary.classic, colors.secondary.pale]
