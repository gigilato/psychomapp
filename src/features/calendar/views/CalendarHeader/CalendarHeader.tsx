import { Canvas, LinearGradient, Rect, vec } from '@shopify/react-native-skia'
import clsx from 'clsx'
import { StyleSheet, useWindowDimensions } from 'react-native'

import { Box, Center, HStack, Text, Pressable } from '$atoms'
import { dayOfWeekFormat, format, monthFormat } from '$libs/dates'
import { PressableIcon } from '$molecules'
import { useSafeAreaInsets } from '$views/libs/safeAreaInsets'

import { interval, textColor, date, gradient, height } from './CalendarHeader.lib'

export const CalendarHeader = () => {
  const { width } = useWindowDimensions()
  const { top } = useSafeAreaInsets()

  return (
    <Box shadow className="rounded-b-[48px] mb-xs" style={{ height }}>
      <Box className="rounded-b-[48px] overflow-hidden" style={{ height }}>
        <Canvas style={StyleSheet.absoluteFill}>
          <Rect x={0} y={0} width={width} height={height}>
            <LinearGradient start={vec(0, 0)} end={vec(width, height)} colors={gradient} />
          </Rect>
        </Canvas>
        <Box className="flex-1 px-l">
          <HStack className="justify-between items-center" style={{ paddingTop: top + 12 }}>
            <Text variant="title" className={textColor} textTransform="upper-first">
              {format(date, monthFormat)}
            </Text>
            <PressableIcon ID="" icon="plus-circle" iconSize={30} iconClassName={textColor} />
          </HStack>
          <HStack className="items-center flex-1">
            {interval.map((item) => {
              const isToday = item.getDate() === date.getDate()
              return (
                <Center key={item.getTime()} className="flex-1">
                  <Text className={textColor}>{format(item, dayOfWeekFormat)}</Text>
                  <Pressable
                    ID=""
                    className={clsx(
                      'w-[40] h-[40] rounded-full mt-s items-center justify-center',
                      isToday ? 'bg-white-classic' : ''
                    )}
                  >
                    <Text
                      variant="blackBody"
                      className={clsx(textColor, isToday ? 'text-primary-strong' : '')}
                    >
                      {item.getDate()}
                    </Text>
                  </Pressable>
                </Center>
              )
            })}
          </HStack>
        </Box>
      </Box>
    </Box>
  )
}
