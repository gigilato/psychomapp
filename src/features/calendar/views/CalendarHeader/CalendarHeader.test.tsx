import { render } from '@testing-library/react-native'

import { CalendarHeader } from './CalendarHeader'

jest.mock('./CalendarHeader.lib', () => {
  return {
    ...jest.requireActual('./CalendarHeader.lib'),
    date: new Date(2024, 0, 3),
    interval: Array.from({ length: 7 }, (_, i) => new Date(2024, 0, i + 1)),
  }
})

describe('CalendarHeader', () => {
  it('should render correctly', async () => {
    const Component = render(<CalendarHeader />)
    expect(Component).toMatchSnapshot()
  })
})
