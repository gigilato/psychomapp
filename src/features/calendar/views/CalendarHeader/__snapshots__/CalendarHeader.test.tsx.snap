// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarHeader should render correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFCF9",
      },
      {
        "shadowColor": "rgba(0, 0, 0, 0.1)",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 6,
      },
      {
        "shadowColor": "#D3D3D3",
      },
      {
        "borderBottomLeftRadius": 48,
        "borderBottomRightRadius": 48,
      },
      {
        "marginBottom": 8,
      },
      [
        undefined,
        {
          "height": 250,
        },
      ],
    ]
  }
>
  <View
    style={
      [
        {
          "borderBottomLeftRadius": 48,
          "borderBottomRightRadius": 48,
        },
        {
          "overflow": "hidden",
        },
        [
          undefined,
          {
            "height": 250,
          },
        ],
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        height={250}
        width={750}
        x={0}
        y={0}
      >
        <View
          colors={
            [
              "#BB99DD",
              "#D8BFD8",
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        [
          {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          {
            "paddingLeft": 24,
            "paddingRight": 24,
          },
          [
            undefined,
            undefined,
          ],
        ]
      }
    >
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "justifyContent": "space-between",
            },
            {
              "alignItems": "center",
            },
            [
              undefined,
              {
                "paddingTop": 28,
              },
            ],
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "SpaceGrotesk-Bold",
              },
              {
                "fontSize": 34,
              },
              {
                "lineHeight": 42,
              },
              {
                "color": "#FFFCF9",
              },
            ]
          }
        >
          Janvier
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={5}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              [
                {
                  "opacity": undefined,
                },
                undefined,
              ],
            ]
          }
          testID=""
        >
          <Text />
        </View>
      </View>
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "alignItems": "center",
            },
            {
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
            },
            [
              undefined,
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            [
              {
                "justifyContent": "center",
              },
              {
                "alignItems": "center",
              },
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
              },
              [
                undefined,
                undefined,
              ],
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontFamily": "SpaceGrotesk-Regular",
                },
                {
                  "fontSize": 16,
                },
                {
                  "lineHeight": 18,
                },
                {
                  "color": "#FFFCF9",
                },
              ]
            }
          >
            lu.
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "paddingBottom": 12,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 12,
                },
                {
                  "borderBottomLeftRadius": 9999,
                  "borderBottomRightRadius": 9999,
                  "borderTopLeftRadius": 9999,
                  "borderTopRightRadius": 9999,
                },
                {
                  "marginTop": 12,
                },
                [
                  {
                    "opacity": undefined,
                  },
                  undefined,
                ],
              ]
            }
            testID=""
          >
            <Text
              style={
                [
                  {
                    "fontFamily": "SpaceGrotesk-Bold",
                  },
                  {
                    "fontSize": 16,
                  },
                  {
                    "lineHeight": 18,
                  },
                  {
                    "color": "#FFFCF9",
                  },
                ]
              }
            >
              1
            </Text>
          </View>
        </View>
        <View
          style={
            [
              {
                "justifyContent": "center",
              },
              {
                "alignItems": "center",
              },
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
              },
              [
                undefined,
                undefined,
              ],
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontFamily": "SpaceGrotesk-Regular",
                },
                {
                  "fontSize": 16,
                },
                {
                  "lineHeight": 18,
                },
                {
                  "color": "#FFFCF9",
                },
              ]
            }
          >
            ma.
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "paddingBottom": 12,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 12,
                },
                {
                  "borderBottomLeftRadius": 9999,
                  "borderBottomRightRadius": 9999,
                  "borderTopLeftRadius": 9999,
                  "borderTopRightRadius": 9999,
                },
                {
                  "marginTop": 12,
                },
                [
                  {
                    "opacity": undefined,
                  },
                  undefined,
                ],
              ]
            }
            testID=""
          >
            <Text
              style={
                [
                  {
                    "fontFamily": "SpaceGrotesk-Bold",
                  },
                  {
                    "fontSize": 16,
                  },
                  {
                    "lineHeight": 18,
                  },
                  {
                    "color": "#FFFCF9",
                  },
                ]
              }
            >
              2
            </Text>
          </View>
        </View>
        <View
          style={
            [
              {
                "justifyContent": "center",
              },
              {
                "alignItems": "center",
              },
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
              },
              [
                undefined,
                undefined,
              ],
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontFamily": "SpaceGrotesk-Regular",
                },
                {
                  "fontSize": 16,
                },
                {
                  "lineHeight": 18,
                },
                {
                  "color": "#FFFCF9",
                },
              ]
            }
          >
            me.
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "paddingBottom": 12,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 12,
                },
                {
                  "borderBottomLeftRadius": 9999,
                  "borderBottomRightRadius": 9999,
                  "borderTopLeftRadius": 9999,
                  "borderTopRightRadius": 9999,
                },
                {
                  "marginTop": 12,
                },
                {
                  "backgroundColor": "#FFFCF9",
                },
                [
                  {
                    "opacity": undefined,
                  },
                  undefined,
                ],
              ]
            }
            testID=""
          >
            <Text
              style={
                [
                  {
                    "fontFamily": "SpaceGrotesk-Bold",
                  },
                  {
                    "fontSize": 16,
                  },
                  {
                    "lineHeight": 18,
                  },
                  {
                    "color": "#FFFCF9",
                  },
                  {
                    "color": "#FFA500",
                  },
                ]
              }
            >
              3
            </Text>
          </View>
        </View>
        <View
          style={
            [
              {
                "justifyContent": "center",
              },
              {
                "alignItems": "center",
              },
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
              },
              [
                undefined,
                undefined,
              ],
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontFamily": "SpaceGrotesk-Regular",
                },
                {
                  "fontSize": 16,
                },
                {
                  "lineHeight": 18,
                },
                {
                  "color": "#FFFCF9",
                },
              ]
            }
          >
            je.
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "paddingBottom": 12,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 12,
                },
                {
                  "borderBottomLeftRadius": 9999,
                  "borderBottomRightRadius": 9999,
                  "borderTopLeftRadius": 9999,
                  "borderTopRightRadius": 9999,
                },
                {
                  "marginTop": 12,
                },
                [
                  {
                    "opacity": undefined,
                  },
                  undefined,
                ],
              ]
            }
            testID=""
          >
            <Text
              style={
                [
                  {
                    "fontFamily": "SpaceGrotesk-Bold",
                  },
                  {
                    "fontSize": 16,
                  },
                  {
                    "lineHeight": 18,
                  },
                  {
                    "color": "#FFFCF9",
                  },
                ]
              }
            >
              4
            </Text>
          </View>
        </View>
        <View
          style={
            [
              {
                "justifyContent": "center",
              },
              {
                "alignItems": "center",
              },
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
              },
              [
                undefined,
                undefined,
              ],
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontFamily": "SpaceGrotesk-Regular",
                },
                {
                  "fontSize": 16,
                },
                {
                  "lineHeight": 18,
                },
                {
                  "color": "#FFFCF9",
                },
              ]
            }
          >
            ve.
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "paddingBottom": 12,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 12,
                },
                {
                  "borderBottomLeftRadius": 9999,
                  "borderBottomRightRadius": 9999,
                  "borderTopLeftRadius": 9999,
                  "borderTopRightRadius": 9999,
                },
                {
                  "marginTop": 12,
                },
                [
                  {
                    "opacity": undefined,
                  },
                  undefined,
                ],
              ]
            }
            testID=""
          >
            <Text
              style={
                [
                  {
                    "fontFamily": "SpaceGrotesk-Bold",
                  },
                  {
                    "fontSize": 16,
                  },
                  {
                    "lineHeight": 18,
                  },
                  {
                    "color": "#FFFCF9",
                  },
                ]
              }
            >
              5
            </Text>
          </View>
        </View>
        <View
          style={
            [
              {
                "justifyContent": "center",
              },
              {
                "alignItems": "center",
              },
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
              },
              [
                undefined,
                undefined,
              ],
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontFamily": "SpaceGrotesk-Regular",
                },
                {
                  "fontSize": 16,
                },
                {
                  "lineHeight": 18,
                },
                {
                  "color": "#FFFCF9",
                },
              ]
            }
          >
            sa.
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "paddingBottom": 12,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 12,
                },
                {
                  "borderBottomLeftRadius": 9999,
                  "borderBottomRightRadius": 9999,
                  "borderTopLeftRadius": 9999,
                  "borderTopRightRadius": 9999,
                },
                {
                  "marginTop": 12,
                },
                [
                  {
                    "opacity": undefined,
                  },
                  undefined,
                ],
              ]
            }
            testID=""
          >
            <Text
              style={
                [
                  {
                    "fontFamily": "SpaceGrotesk-Bold",
                  },
                  {
                    "fontSize": 16,
                  },
                  {
                    "lineHeight": 18,
                  },
                  {
                    "color": "#FFFCF9",
                  },
                ]
              }
            >
              6
            </Text>
          </View>
        </View>
        <View
          style={
            [
              {
                "justifyContent": "center",
              },
              {
                "alignItems": "center",
              },
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
              },
              [
                undefined,
                undefined,
              ],
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontFamily": "SpaceGrotesk-Regular",
                },
                {
                  "fontSize": 16,
                },
                {
                  "lineHeight": 18,
                },
                {
                  "color": "#FFFCF9",
                },
              ]
            }
          >
            di.
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "paddingBottom": 12,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 12,
                },
                {
                  "borderBottomLeftRadius": 9999,
                  "borderBottomRightRadius": 9999,
                  "borderTopLeftRadius": 9999,
                  "borderTopRightRadius": 9999,
                },
                {
                  "marginTop": 12,
                },
                [
                  {
                    "opacity": undefined,
                  },
                  undefined,
                ],
              ]
            }
            testID=""
          >
            <Text
              style={
                [
                  {
                    "fontFamily": "SpaceGrotesk-Bold",
                  },
                  {
                    "fontSize": 16,
                  },
                  {
                    "lineHeight": 18,
                  },
                  {
                    "color": "#FFFCF9",
                  },
                ]
              }
            >
              7
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
